src/extension.ts

```diff
@@ -221,9 +220,25 @@ export async function activate(context: ExtensionContext): Promise<PythonEnviron
             traceInfo(
                 `Internal: Changed environment from ${e.old?.displayName} to ${e.new?.displayName} for: ${location}`,
             );
-            await updateActivateMenuButtonContext(terminalManager, projectManager, envManagers);
             updateViewsAndStatus(statusBar, workspaceView, managerView, api);
         }),
+        onDidChangeTerminalShellIntegration(async (e) => {
+            const envVar = e.shellIntegration?.env;
+            if (envVar) {
+                if (envVar['VIRTUAL_ENV']) {
```

Comment:
VIRTUAL_ENV would be indicative only for `venv` and `virtualenv`, right?
I assume there are other things we need to look for remaining virtual environments, which may be up to the virtual env providers.

Reply:
That is correct, we plan on support `venv` built-in, we will likely expose an API for env managers to provide this data.